src/server.js

```diff
@@ -76,6 +76,48 @@ export class Server {
 
   /**
    * Submits a transaction to the network.
+   *
+   * Warning: sometimes a transaction will "succeed" (without throwing an error)
+   * but the transaction won't appear on the network.
+   *
+   * For example, a user submits a `manageOffer` transaction that would
+   * result in a user receiving less than 0.0000001 of a token. The transaction
+   * won't be logged on the network, yet `submitTransaction` will resolve to this:
+   *
+   * ```json
+   * {
+   *   "_links": {
+   *     "transaction": {
+   *       "href": "https://horizon.stellar.org/transactions/f765724bcf5df1b44f0899ec0287ec7a994e4a98505b819453fee3b39642d185"
+   *     }
+   *   },
+   *   "hash": "f765724bcf5df1b44f0899ec0287ec7a994e4a98505b819453fee3b39642d185",
+   *   "ledger": 22781293,
+   *   "envelope_xdr": "AAAAAIUAEW3jQt3+fbT6nCASA1/8RWdp9fJ2woxqPHZPQUH/AAAAZAEH/OgAAAAXAAAAAQAAAAAAAAAAAAAAAFx+o4kAAAAAAAAAAQAAAAAAAAADAAAAAAAAAAFCQVQAAAAAAEZK09vHmzOmEMoVWYtbbZcKv3ZOoo06ckzbhyDIFKfhAAAAAAAAAAEAACcQAABOHwAAAAAAAAAAAAAAAAAAAAFPQUH/AAAAQB4Ckk7T0OTf1M4otquqdJb7iyDZf7dGS3m39HevsHTKV2eYqc9921sWELh75E7zfNWcuwg+z5MCedhgCTIqLQI=",
+   *   "result_xdr": "AAAAAAAAAGQAAAAAAAAAAQAAAAAAAAADAAAAAAAAAAAAAAACAAAAAA==",
+   *   "result_meta_xdr": "AAAAAQAAAAIAAAADAVudbQAAAAAAAAAAhQARbeNC3f59tPqcIBIDX/xFZ2n18nbCjGo8dk9BQf8AAAACVJZtHAEH/OgAAAAWAAAABgAAAAEAAAAAhD8BLsZFQEF33rKS6YopQUT3b6iLBG4nspe68/DBNBYAAAAAAAAAAAEAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEBW51tAAAAAAAAAACFABFt40Ld/n20+pwgEgNf/EVnafXydsKMajx2T0FB/wAAAAJUlm0cAQf86AAAABcAAAAGAAAAAQAAAACEPwEuxkVAQXfespLpiilBRPdvqIsEbieyl7rz8ME0FgAAAAAAAAAAAQAAAAAAAAAAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAAA="
+   * }
+   * ```
+   *
+   * To catch failed transactions like this, you should inspect the resulting
+   * XDR object to see if it's an error!
+   *
+   * ```javascript
+   * const result = await StellarSdk.StellarServer.submitTransaction(transaction);
+   *
+   * // if the transaction is valid, it'll be a real TransactionEnvelope
+   * try {
+   *  const resultXDR = StellarSdk.xdr.TransactionEnvelope.fromXDR(result.result_xdr, 'base64');
```

Comment:
Why `xdr.TransactionEnvelope`? Should be `xdr.TransactionResult`.
And the `TransactionResult` class contains a switch which can tell whether the transaction succeeded or not (see https://github.com/stellar/stellar-core/blob/master/src/xdr/Stellar-transaction.x#L750). 
```
const resultXDR = StellarSdk.xdr.TransactionResult.fromXDR(result.result_xdr, 'base64');
if (resultXDR.result().switch().value<0) {
  //tx failed, the value returned by the switch can be decoded using TransactionResultCode enum
} else {
  //success
}
```

Reply:
The SDK can handle this situation internally and return `Promise.reject()` in the case of failed tx, right? Or maybe I'm missing something?

Reply:
Thanks for pointing this out, I had no idea. 

As for the SDK rejecting: that's probably the better user experience, but to preserve backwards compatibility it's probably best to keep the behavior the same, and to give developers fair warning about what's actually happening.

Reply:
I have never checked the result XDR myself, and doubt that any of hundreds of existing wallets/apps check it once the promise was successfully resolved. Fixing this bug on the SDK level will save tons of work for the ecosystem developers because literally all applications that submit transactions using SDK will have to adopt the same workaround. Release it in 0.15 version and state that behavior explicitly in the changelog if you like, but please don't force all JS devs to copy-paste the same workaround code in their application.

Reply:
That makes sense.

Reply:
We shouldn't do it, sorry. Transactions can have multiple operations. What if you submit 2 `manage_offer`s and one is actually executed and the other is not and is deleted. Rejecting the promise would be extra confusing and could lead to someone submitting this transaction twice or more times.

Not sure what's the best option right now (maybe another method that help user get data from tx result to better understand it?). Should discuss this in another issue (or reopen #209)?

Reply:
I thought either all operations within a transaction succeed, or they all fail?

Reply:
I misunderstood #209. I thought the transactions were failing on the network but the SDK was reporting them as a success. In actually, the transactions are succeeding completely, but the offer is getting marked as deleted immediately.